refactor/nfs: update NFS API to use symmetric encryption

nbaksalyar opened the pull request:
Update NFS metadata structs to reflect changes proposed in
https://gist.github.com/ustulation/b1009a943ac9deed5331b5d5d1003f20

Commit:
refactor/nfs: update NFS API to use symmetric encryption

Update NFS metadata structs to reflect changes proposed in
https://gist.github.com/ustulation/b1009a943ac9deed5331b5d5d1003f20

ustulation commented on src/nfs/dir.rs:
```diff
@@ -0,0 +1,235 @@
+}
+
+#[cfg(test)]
+mod test {
```
These are to be named plural as per new rule in maidsafe repos - so `mod tests`

ustulation commented on src/nfs/metadata/dir_metadata.rs:
```diff
@@ -0,0 +1,172 @@
+    fn update_using_setters() {
+        let modified_time = ::time::now_utc();
+        let id = rand::random();
+        let mut obj_before = DirMetadata::new(id, "hello.txt".to_string(), Vec::new(), None);
```
Since it's into-string now - so might not need `to_string()` here anymore

ustulation commented on src/nfs/metadata/dir_metadata.rs:
```diff
@@ -0,0 +1,172 @@
+        obj_before.set_name("index.txt".to_string());
+        let serialised_data = unwrap!(serialise(&obj_before));
+        let obj_after: DirMetadata = unwrap!(deserialise(&serialised_data));
+        assert_eq!(*user_metadata, *obj_after.user_metadata());
```
Since it implements `Eq and PartialEq` maybe simply use `assert_eq!(dir_meta_before, dir_meta_after)` instead of for each field

ustulation commented on src/nfs/metadata/mod.rs:
```diff
@@ -16,8 +16,9 @@
-pub mod directory_metadata;
+mod file_metadata;
+/// DirMetadata
+mod dir_metadata;
```
each grouping should be alphabetical - so rustfmt does it for imports but we will need to do it manually for types/mod/extern-crates etc (or if there is an option now in rustfmt, pls updated the rustfmt.toml and check it in too).

ustulation commented on src/nfs/mod.rs:
```diff
@@ -19,21 +19,21 @@
-/// Module for directory reltaed structs - DirectoryListin, DirectoryInfo
-pub mod directory_listing;
+/// Module for directory related structs
+pub mod dir;
```
needs to be alphabetical

Commit:
style/rustfmt: fix formatting

Commit:
style/core: apply rustfmt to mock routing

Commit:
refactor/core: update client/account to symmetric encryption

Applying changes from this branch: https://github.com/madadam/safe_core/tree/nfs-symmetric-keys
